env_test.go

```diff
@@ -2077,3 +2077,48 @@ func TestIssue308(t *testing.T) {
 	isNoErr(t, Parse(&cfg))
 	isEqual(t, Issue308Map{"FOO": []string{"BAR", "ZAZ"}}, cfg.Inner)
 }
+
+func TestIssue298(t *testing.T) {
+	type Test struct {
+		Str string `env:"STR"`
+		Num int    `env:"NUM"`
+	}
+	type ComplexConfig struct {
+		Foo *[]Test `envPrefix:"FOO_"`
+		Bar []Test  `envPrefix:"BAR_"`
+		Baz *Test
+	}
+
+	t.Setenv("FOO_0_STR", "f0t")
+	t.Setenv("FOO_0_NUM", "101")
+	t.Setenv("FOO_1_STR", "f1t")
```

Comment:
if not setting `FOO_0_NUM` ,just setting `FOO_1_NUM`, default value for `FOO_0_NUM` is empty? WDYT?

Reply:
Yes, The value will be left empty, unless the Foo[0].Num has assigned a value before

Reply:
Also, there is a point, The number sequence should be followed from 0 as below:
`FOO_0_STR=some-str`
`FOO_1_NUM=123`

So only assigning `FOO_1_NUM` or `FOO_1000_NUM` without any `FOO_0_` up to that number, will not work and will be ignored

Reply:
Got it. Thanks.